app/helpers/perilune/home_helper.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Perilune
+  module HomeHelper
+    def get_general_success_and_failure_series(type:, from:, to:)
+      success_series = get_general_series_for(
+        type: type, from: from,
+        to: to, status: 'success'
+      )
+
+      failure_series = get_general_series_for(
+        type: type, from: from,
+        to: to, status: 'failure'
+      )
+
+      [success_series, failure_series]
+    end
+
+    private
+
+    def get_general_series_for(type:, from:, to:, status:)
+      series = Trifle::Stats.series(
+        key: "perilune::#{type}::#{status}", from: from, to: to,
+        range: :hour, skip_blanks: true
+      )
+
+      series.transpond.average(
+        path: type, key: 'average_duration',
+        sum: 'duration', count: 'count'
+      )
+
+      format_timeline_data(series, "#{type}.average_duration")
+    end
```

Comment:
Here's the private method.

As said before, it searches with our data into Mongo with Trifle gem. Also, we apply transponding (average operation) on all of our series. We use the duration value and the count to transpond our data.

Finally, we call format_timeline_data to prepare data structure for Highchart.